### PR TITLE
Validate string as utf8 in Font::DetermineLines

### PR DESCRIPTION
**Pull request by dbenage-cx**
Workaround for #2166

### CODE REVIEW

File: GG/src/Font.cpp [hunk 1]
```diff
@@ -1668,6 +1668,13 @@ std::vector<Font::LineData> Font::DetermineLines(
     const std::string& text, Flags<TextFormat>& format, X box_width,
     const std::vector<std::shared_ptr<TextElement>>& text_elements) const
 {
+    // HACK - Workaround for #2166
+    // On OSX, right clicking an unowned planet at game start may result in utf8::invalid_utf8 or utf8::not_enough_room
+    if (!utf8::is_valid(text.begin(), text.end())) {
+        std::cout << "Invalid UTF8 in text: " << text;
```
Comment (geoffthemedio) [inline GG/src/Font.cpp]: Use `ErrorLogger()` not `std::cout`.

Comment (dbenage-cx) [reply GG/src/Font.cpp]: It might be beneficial to rework loggers into GiGi/as seperate lib later, currently they are not accessible.

**Comment by geoffthemedio** [comment]
Then at least use `std::cerr`

**Comment by Vezzra** [comment]
@dbenage-cx, bump? Can you make the adjustment @geoffthemedio requested? So we can get this in...

**Comment by geoffthemedio** [comment]
@Vezzra He already did.

**Comment by Vezzra** [comment]
Oops, sorry, that happens when I just hurry over stuff... 😔

**Review by Vezzra — APPROVED** [review]

**Comment by Vezzra** [comment]
Anyone any objections against merging this?

**Pull request merged**